### PR TITLE
MAC address recalculation

### PR DESCRIPTION
**Pull request by LostFuse**
* While swapping an SD card with the system to another platform the MAC address persisted from the previous one. 
  To prevent that, force MAC recalculation on every boot was enabled.
* Enable overriding environment variables from U-Boot shell for manual MAC
  address modification.

### CODE REVIEW

File: recipes-bsp/u-boot/u-boot/0004-sun8i-disable-Vendor-Parameter-Protection.patch [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+diff --git a/include/configs/sun8i.h b/include/configs/sun8i.h
+index 9b4675e4c374..228254e2076d 100644
+--- a/include/configs/sun8i.h
++++ b/include/configs/sun8i.h
+@@ -12,6 +12,8 @@
+  * A23 specific configuration
+  */
+ 
++#define CONFIG_ENV_OVERWRITE 1
```
Comment (tomzy-0) [inline recipes-bsp/u-boot/u-boot/0004-sun8i-disable-Vendor-Parameter-Protection.patch:22]: @leckijakub A short comment explaining why we are setting this config would be nice

File: recipes-bsp/u-boot/u-boot/0005-generate-random-ethaddr-by-hardcoding-invalid-addres.patch [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ #else /* ifndef CONFIG_SPL_BUILD */
+diff --git a/net/eth-uclass.c b/net/eth-uclass.c
+index ed81cbd537..ba64446563 100644
+--- a/net/eth-uclass.c
++++ b/net/eth-uclass.c
+@@ -551,6 +551,7 @@ static int eth_post_probe(struct udevice *dev)
+ 		net_random_ethaddr(pdata->enetaddr);
+ 		printf("\nWarning: %s (eth%d) using random MAC address - %pM\n",
+ 		       dev->name, dev->seq, pdata->enetaddr);
++		eth_env_set_enetaddr_by_index("eth", dev->seq, pdata->enetaddr);
```
Comment (tomzy-0) [inline recipes-bsp/u-boot/u-boot/0005-generate-random-ethaddr-by-hardcoding-invalid-addres.patch]: @leckijakub Let's add some comments also here, maybe a small printf informing that ethaddr was saved in the environment variable

File: recipes-bsp/u-boot/u-boot/0004-sun8i-disable-Vendor-Parameter-Protection.patch (hunk 1 above)
Comment (LostFuse) [reply recipes-bsp/u-boot/u-boot/0004-sun8i-disable-Vendor-Parameter-Protection.patch:22]: @TomaszAIR I have added a comment

File: recipes-bsp/u-boot/u-boot/0005-generate-random-ethaddr-by-hardcoding-invalid-addres.patch (hunk 2 above)
Comment (LostFuse) [reply recipes-bsp/u-boot/u-boot/0005-generate-random-ethaddr-by-hardcoding-invalid-addres.patch]: @TomaszAIR comment along with a `printf` added

**Pull request merged**